Fixing Init test for Serial only

aprokop opened the pull request.

aprokop commented:
In `Kokkos_Core_fwd.hpp` we have
```c++
#if   defined( KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_CUDA )
  typedef Cuda DefaultExecutionSpace;
#elif defined ( KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_OPENMPTARGET )
  typedef Experimental::OpenMPTarget DefaultExecutionSpace ;
#elif defined( KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_OPENMP )
  typedef OpenMP DefaultExecutionSpace;
#elif defined( KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_THREADS )
  typedef Threads DefaultExecutionSpace;
//#elif defined( KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_QTHREADS )
//  typedef Qthreads DefaultExecutionSpace;
#elif defined( KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_SERIAL )
  typedef Serial DefaultExecutionSpace;
#else
#  error "At least one of the following execution spaces must be defined in order to use Kokkos: Kokkos::Cuda, Kokkos::Experimental::OpenMPTarget, Kokkos::OpenMP, Kokkos::Threads, Kokkos::Qthreads, or Kokkos::Serial."
#endif
```

Thus, the only way `Serial` is the default execution space is when all other spaces are disabled. So it is fine to leave the current code in `DTK_Core.cpp`.

Rombur commented on packages/Interface/test/tstInit.cpp:
```diff
@@ -40,47 +58,38 @@ int main( int argc, char *argv[] )
     {
-        check( !DataTransferKit::isInitialized() &&
-               !Kokkos::DefaultExecutionSpace::is_initialized() );
+        check( !DataTransferKit::isInitialized() && !isKokkosInitialized() );
```
`isKokkosInitialized()` is true -> the test fails

aprokop commented:
> In gdb, it looks like there is a problem with getting typeid(default_space) (no symbol for default_space and no symbol for DefaultExecutionSpace in Kokkos

This is a red herring. You are right in the sense that gdb does not show those symbols. But if you add printing `typeid(default_space).name()` in the actual code, you can see it.

The logic in the PR was flawed due to my stupidity. I tried to make the code easy to read, and moved the code from `main()` to a separate function. That was a mistake, as you need to take negative of only one clause in that function.

I updated the PR with the right logic, and tested it on Serial. Should work for you too.
